### PR TITLE
Add username/password auth for tokens

### PR DESCRIPTION
**Pull request by Tschrock**
- Rewrote `getToken` to use `request` instead of `jsdom` and added a `cookieJar` parameter.
 - Added a `getAuthedCookieJar` function which gets a logged-in cookieJar for a username/password.
 - Added a `getAuthedToken` function that gets a logged-in chat token for a username/password.

### CODE REVIEW

File: modules/picarto.js [hunk 1]
```diff
@@ -107,27 +107,80 @@ function getChannel(stream) {
...
 			}
 		});
-	})
+	});
+}
+
+function getAuthedCookieJar(username, password) {
+	var cookieJar = request.jar();
+	return new Promise(function (resolve, reject) {
+		console.log("getAuthedCookieJar: ", username, password);
```
Comment (Wolvan) [inline modules/picarto.js]: Don't really like the fact that username and password gets printed here

File: modules/picarto.js [hunk 2]
```diff
@@ -1,4 +1,4 @@
-﻿'use strict';
+'use strict';
```
Comment (Wolvan) [inline modules/picarto.js:1]: Unsure of what exactly caused this line to be modified

File: modules/picarto.js [hunk 3]
```diff
@@ -159,6 +212,7 @@ function picarto_connection(_token, _debug) {
 	};
 }
 
+
```
Comment (Wolvan) [inline modules/picarto.js]: Mind removing this newline? It's just cosmetics, but something that you could do while you are editing the file anyways, if you don't mind.

**Review by Wolvan — CHANGES_REQUESTED** [review]
Have not tested it yet, gonna do so tomorrow but from what I can see at 3 in the morning after being almost 24h awake it looks good to me, apart from the small cosmetic changes I listed.

File: modules/picarto.js (hunk 2 above)
Comment (Tschrock) [reply modules/picarto.js:1]: That would be me removing the utf-8 BOM, which in addition to being completely unnecessary, adds an extra white-space and shows as a syntax error in Netbeans :/
From what I can see, half of your files have it and half of them dont.

File: modules/picarto.js (hunk 1 above)
Comment (Tschrock) [reply modules/picarto.js]: :face_with_head_bandage:  That would be me coding in the middle of the night and forgetting to remove one of my debugging checks (I did test this you know :wink: )

File: modules/picarto.js (hunk 3 above)
Comment (Tschrock) [reply modules/picarto.js]: :+1:

File: modules/picarto.js (hunk 2 above)
Comment (Wolvan) [reply modules/picarto.js:1]: Well, fuck me then. Figured that it must have been some kind of invisible utf-8 control character thingie. It's cool, thanks for removing that.

File: modules/picarto.js (hunk 3 above)
Comment (Wolvan) [reply modules/picarto.js]: Thanks m8

File: modules/picarto.js (hunk 1 above)
Comment (Wolvan) [reply modules/picarto.js]: I am sure you did test it! Doesn't mean that I won't test it myself again :wink:. But yeah, it's the only debug command I found, so removing that should make it OK

**Review by Wolvan — APPROVED** [review]

**Comment by Wolvan** [comment]
Neat, works perfectly. Thanks a lot and sorry for taking so long to merge the branch in!

**Pull request merged**